curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -1124,37 +1213,55 @@ std::string DataCache::GenerateObjectName(uint64_t chunkId,
     return oss.str();
 }
 
-CURVEFS_ERROR DataCache::Flush(Inode* inode, bool force) {
+CURVEFS_ERROR DataCache::Flush(uint64_t inodeId, bool force) {
     uint64_t blockSize = s3ClientAdaptor_->GetBlockSize();
     uint64_t chunkSize = s3ClientAdaptor_->GetChunkSize();
-    uint32_t flushInterval = s3ClientAdaptor_->GetFlushInterval();
-    uint64_t fsId = inode->fsid();
-    uint64_t blockPos = chunkPos_ % blockSize;
-    uint64_t blockIndex = chunkPos_ / blockSize;
+    uint32_t flushIntervalSec = s3ClientAdaptor_->GetFlushInterval();
+    uint32_t fsId = s3ClientAdaptor_->GetFsId();
+    uint64_t blockPos;
+    uint64_t blockIndex;
     uint64_t chunkIndex = chunkCacheManager_->GetIndex();
-    uint64_t offset = chunkIndex * chunkSize + chunkPos_;
-    uint64_t tmpLen = len_;
+    uint64_t offset;
+    uint64_t tmpLen;
     uint64_t n = 0;
     std::string objectName;
     uint32_t writeOffset = 0;
     bool isFlush = true;
     uint64_t chunkId;
     uint64_t now = ::curve::common::TimeUtility::GetTimeofDaySec();
+    // std::unique_ptr<char[]> data(new char[len_]);
+    char* data;
+    FSStatusCode ret;
     if (!force) {
-        if (len_ = chunkSize) {
+        if (len_ == chunkSize) {
             isFlush = true;
-        } else if (now < (createTime_ + flushInterval)) {
+        } else if (now < (createTime_ + flushIntervalSec)) {
             isFlush = false;
         }
     }
-
-    CURVEFS_ERROR ret = s3ClientAdaptor_->AllocS3ChunkId(fsId, &chunkId);
-    if (ret != CURVEFS_ERROR::OK) {
-        LOG(ERROR) << "alloc s3 chunkid fail. ret:" << ret;
-        return ret;
-    }
+    LOG(INFO) << "now:" << now << ",createTime:" << createTime_
+              << ",flushIntervalSec:" << flushIntervalSec
+              << ",isFlush:" << isFlush;
 
     if (isFlush) {
+        {
+            curve::common::LockGuard lg(mtx_);
+            tmpLen = len_;
+            blockPos = chunkPos_ % blockSize;
+            blockIndex = chunkPos_ / blockSize;
+            offset = chunkIndex * chunkSize + chunkPos_;
+
+            data = new char[len_];
+            memcpy(data, data_, len_);
```

Comment:
what's the purpose of data, and you don't use it when update data to s3 or diskcache(line: 1277-1283)

Reply:
Why memcpy data_ to data, and you dons't  use data any way?